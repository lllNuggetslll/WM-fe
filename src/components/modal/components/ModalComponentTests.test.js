import React from "react";
import { shallow } from "enzyme";
import { BusinessAddress, BusinessHours, ModalHeader } from "./";

describe("Modal components", () => {
  const listing = {
    address: "address",
    state: "state",
    zip_code: 1,
    city: "city",
    phone_number: 1
  };

  describe("BusinessAddress", () => {
    it("should render", () => {
      const wrapper = shallow(<BusinessAddress {...{ listing }} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("BusinessHours", () => {
    it("should render", () => {
      const hours = {
        sat: {
          open: " 6am",
          close: "8pm"
        }
      };
      const wrapper = shallow(<BusinessHours {...{ hours }} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("ModalHeader", () => {
    it("should render", () => {
      const onCloseModal = () => {};
      const wrapper = shallow(<ModalHeader {...{ listing, onCloseModal }} />);

      expect(wrapper).toMatchSnapshot();
    });

    it("should fire onClick on click", () => {
      const onCloseModal = jest.fn();
      const wrapper = shallow(<ModalHeader {...{ listing, onCloseModal }} />);

      wrapper.find('[data-testid="xIcon"]').simulate("click");

      expect(onCloseModal).toHaveBeenCalled();
    });
  });
});
