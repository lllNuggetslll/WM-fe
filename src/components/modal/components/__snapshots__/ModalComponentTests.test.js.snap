// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal components BusinessAddress should render 1`] = `
<styled.div>
  <div>
    address
  </div>
  <div>
    <span>
      city
      , 
    </span>
    <span>
      state
      , 
    </span>
    <span>
      1
    </span>
  </div>
  <div>
    1
  </div>
</styled.div>
`;

exports[`Modal components BusinessHours should render 1`] = `
<styled.table>
  <tr>
    <th>
      Day
    </th>
    <th>
      Opens
    </th>
    <th>
      Closes
    </th>
  </tr>
  <tr
    key="Sat"
  >
    <td>
      Sat
    </td>
    <td>
       6am
    </td>
    <td>
      8pm
    </td>
  </tr>
</styled.table>
`;

exports[`Modal components ModalHeader should render 1`] = `
<styled.div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Avatar
      height={100}
      img="undefined"
      width={100}
    />
    <styled.div>
      <styled.div>
        <CardDetails
          listing={
            Object {
              "address": "address",
              "city": "city",
              "phone_number": 1,
              "state": "state",
              "zip_code": 1,
            }
          }
        />
      </styled.div>
    </styled.div>
  </div>
  <styled.div>
    <i
      className="fas fa-times"
      data-testid="xIcon"
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;
